test/dlc_tests/container_tests/bin/pytorch_tests/testTorchdataDev

```diff
@@ -4,13 +4,36 @@ HOME_DIR=/test
 BIN_DIR=${HOME_DIR}/bin
 LOG_DIR=${HOME_DIR}/logs
 
-TORCHDATA_DEV_BRANCH_PREFIX="torchdata_pt_"
-TORCHDATA_DEV_REPO="https://github.com/ShiboXing/data.git"
-TORCHDATA_TEST_SCRIPT="data/test/test_remote_io.py"
-VERSION=$(python -c "import torch; print(torch.__version__.split('+')[0])")
+TORCHDATA_REPO="https://github.com/pytorch/data.git"
+TORCHDATA_TEST_SCRIPT="test/test_remote_io.py"
+TORCH_VERSION=$(python -c "import torch; print(torch.__version__)")
+TORCHDATA_RELEASE_TAG=$(python -c "import torchdata; torchdata_versions = torchdata.__version__.split('.'); print(f'release/{torchdata_versions[0]}.{torchdata_versions[1]}')")
 
-echo "Fetch Torchdata S3 IO test script."
-git clone -b ${TORCHDATA_DEV_BRANCH_PREFIX}${VERSION} ${TORCHDATA_DEV_REPO}
+## using the commit requires user.email and user.name.. 
+echo adding temp git config user email and name
+git config --global user.email "you@example.com"
+git config --global user.name "Your Name"
+
+echo cloning torchdata repo
+git clone ${TORCHDATA_REPO}
+cd data
+
+## PyTorch 1.11 DLC was built with TorchData 0.4.0 which is meant for PyTorch 1.12 and not PT 1.11
+## This fails the Torchdata 0.4.0 test_remote_io.py at test_http_reader_iterdatapipe due to 
+## Torch not knowing "auth" in 1.11. We need to use TorchData 0.3.0 Branch for testing PyTorch 1.11 DLC
+if [[ ${TORCH_VERSION} == *"1.11"* ]]
+then
+    echo "This is PT 1.11.0, checkout release/0.3"
+    git checkout "release/0.3"
+else
+    echo "checkout ${TORCHDATA_RELEASE_TAG}"
+    git checkout ${TORCHDATA_RELEASE_TAG}
+    
+    echo "Patching test_remote_io for E3 bucket issue"
+    git cherry-pick 2a58c306f11225264a2f432f3d483fca61e9576d --strategy-option theirs
```

Comment:
why do we need to cherrypick these commits? Can you add this in the comments so it's clear - the "echo" cmd currently doesn't have enough information for a user to easily elucidate this

Reply:
Added PRs and details for the patching.